### PR TITLE
[System Properties] LPS-158206 Add new functionality to SystemProperties: get system properties by prefix(this function is needed when moving module framwork properties)

### PR DESCRIPTION
**Pull request by W2jx**
Hi @tinatian @dantewang , I add a new method and unit test in this pr.
LPS-158206 depends on LPS-158207

Detail:  https://github.com/liferay-core-infra/liferay-portal/pull/1192

### CODE REVIEW

**Comment by W2jx** [comment]
Hi @catarineaguiar, Could you please help me check the test results when they finish ? Thanks!

**Comment by catarineaguiar** [comment]
@W2jx no regressions found.

File: portal-kernel/test/unit/com/liferay/portal/kernel/util/SystemPropertiesTest.java [hunk 1]
```diff
@@ -58,6 +69,33 @@ public void testGetProperties() {
...
+	@Test
+	public void testGetPropertiesWithPrefix() {
+		Map<String, String> propertiesWithPrefix =
+			SystemProperties.getProperties(_PREFIX, false);
+
+		Assert.assertTrue(propertiesWithPrefix.isEmpty());
+
+		SystemProperties.set(_TEST_KEY, _TEST_VALUE);
+
+		Map<String, String> propertiesWithoutPrefix =
```
Comment (dantewang) [inline portal-kernel/test/unit/com/liferay/portal/kernel/util/SystemPropertiesTest.java:81]: @W2jx This variable can be inlined;

File: portal-kernel/test/unit/com/liferay/portal/kernel/util/SystemPropertiesTest.java [hunk 2]
```diff
@@ -58,6 +69,33 @@ public void testGetProperties() {
...
+			SystemProperties.getProperties(_PREFIX, true);
+
+		Assert.assertEquals(
+			new HashMapBuilder<String, String>().put(
+				_KEY, _TEST_VALUE
+			).build(),
+			propertiesWithoutPrefix);
+
+		propertiesWithPrefix = SystemProperties.getProperties(_PREFIX, false);
```
Comment (dantewang) [inline portal-kernel/test/unit/com/liferay/portal/kernel/util/SystemPropertiesTest.java:90]: @W2jx You can also inline this assigning.

File: portal-kernel/test/unit/com/liferay/portal/kernel/util/SystemPropertiesTest.java [hunk 3]
```diff
@@ -58,6 +69,33 @@ public void testGetProperties() {
...
 	}
 
+	@Test
+	public void testGetPropertiesWithPrefix() {
+		Map<String, String> propertiesWithPrefix =
+			SystemProperties.getProperties(_PREFIX, false);
+
+		Assert.assertTrue(propertiesWithPrefix.isEmpty());
+
+		SystemProperties.set(_TEST_KEY, _TEST_VALUE);
```
Comment (dantewang) [inline portal-kernel/test/unit/com/liferay/portal/kernel/util/SystemPropertiesTest.java:79]: @W2jx It's still not obvious that `_TEST_KEY` is `_PREFIX` concating `_KEY`. I think it's better to write `_PREFIX + _KEY` as the key here.

**Review by dantewang — CHANGES_REQUESTED** [review]

File: portal-kernel/test/unit/com/liferay/portal/kernel/util/SystemPropertiesTest.java (hunk 3 above)
Comment (W2jx) [reply portal-kernel/test/unit/com/liferay/portal/kernel/util/SystemPropertiesTest.java:79]: I will fix it.

**Comment by W2jx** [comment]
Hi @dantewang , I have resended it.

**Review by dantewang — APPROVED** [review]

**Comment by tinatian** [comment]
@W2jx Please rebase.

Tina.

**Comment by W2jx** [comment]
Hi @tinatian , I have updated the code.

**Comment by tinatian** [comment]
@W2jx Code looks good to me. Wait for https://github.com/liferay-core-infra/liferay-portal/pull/1259 to be merged.

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
@W2jx no regressions found.

**Comment by W2jx** [comment]
Ok @tinatian , I will resend the code based on the master branch when https://github.com/liferay-core-infra/liferay-portal/pull/1259 is merged.

**Pull request closed without merging**